src/area-chart/model/use-chart-model.ts

```diff
@@ -55,6 +58,14 @@ export default function useChartModel<T extends AreaChartProps.DataTypes>({
   const containerRef = useRef<HTMLDivElement>(null);
   const verticalMarkerRef = useRef<SVGLineElement>(null);
 
+  const plotMeasureRef = useRef<SVGLineElement>(null);
+  const [measuredHeight, setHeight] = useState(0);
+  useResizeObserver(
+    () => plotMeasureRef.current,
+    entry => fitHeight && setHeight(entry.borderBoxHeight)
```

Comment:
I am guessing the conditional `fitHeight && setHeight(...)` is the one that improves performance. When false, there are no extra state updates and no-op resize observer does not really cost anything.

Is this correct?

Reply:
Yes, I think this combined with the `fitHeight` if conditionals is giving the desired effect. But we can wait for @pan-kot to confirm.

Reply:
Yes, that is correct. The hooks are still there and the resize observer is created yet the associated state is never updated when fitHeight=false. The performance improvements come from having less nodes being rendered when fitHeight=false.